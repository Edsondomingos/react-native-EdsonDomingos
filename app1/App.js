import { StatusBar } from 'expo-status-bar';
import { StyleSheet, Text, View, Image } from 'react-native';  //Por traz, cria constantes e traz como objeto/classe.
import { backgroundColor } from 'react-native/Libraries/Components/View/ReactNativeStyleAttributes';

export default function App() { // export - fica disponivel para a aplicaÃ§Ã£o. default - tudo o que vem da funÃ§Ã£o vai ser executado
  return ( //dentro do 'return' fica o que vai ser mostrado. Como fica disponivel, precisa retornar.
    /*-Dentre do return fica o 'jsx'. jsx Ã© convertido para a linguagem nativa.
    -view se comporta como uma div. Precisa ter uma view e apenas uma, mas pode ter view dentro de outra.
    -Text se comporta como <p> e <span>. Quebra de linha automatica. Pode colocar Text Dentro de Text e fica na mesma linha .
    -Componentes bÃ¡sicos - Button, TextInput, FiatList,SectionList, ... */
    <View style={styles.container}>
    {/* <View style={{backgroundColor:'#ccc'}}> */}
      <Image
        style={img.perfil}
        source={require('./images/ed.jpg')}
      />
      <Text style={{color:'purple',textAlign:'center'}}>Hello world EdsonâœŒ{'\n'}
        Bem vindo ao seu primeiro app com react native.
      </Text>
      <Text style={[estilo.geral, estilo.teste]}>Bio</Text>
      <View style={view2.campu}>
        <Text>
          ðŸ–¥ Atualmente estudante{'\n'}
          ðŸ”Ž Procurando estÃ¡gio{'\n'}
          ðŸ“² Novo desenvolvedor Mobile{'\n'}
          ðŸ’» Desenvolvedor Web
        </Text>
      </View>
      
      <StatusBar style="auto" />
    </View>
  );
}
/* style:
  Em linha - coloca a style={{propriedade:valor}} - um objeto com objetos dentro
  Em constante - Cria uma constante e coloca um objetos dentro dela. para usar {styles.container}, apenas o objeto.
  constante e Classe - Cria uma constante e cria um objeto stylesheet com as classes que contem um objeto
  ex: 
  const estilo = StyleSheet .create({
    geral: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
    }
  })
  Para declarar 2 - {[estilo.geral, estilo.teste]} - Se for apenas um, declara igual Em constante
*/
/*Tamanho Ã© sempre com px e %
  px - coloca apenas o numero - 50
  % - coloca como string - '50%'
*/
const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'aqua',
    alignItems: 'center',
    justifyContent: 'center',
  },
});

const estilo = StyleSheet.create({
  geral: {
    backgroundColor: '#929',
    height: 30,  
  },
  teste: {
    color: 'pink',
    textTransform: 'uppercase',
    width: '30%',
    textAlign: 'center',
    // fontFamily: 'Verdana',
    fontSize: 15,
    fontWeight: 'bold',
    marginTop: 20,
  }
});

const view2 = StyleSheet.create({
  campu:{
    padding: 20,
    backgroundColor: 'white',
    borderWidth: 1,
    borderStyle: 'dotted',
    borderColor: 'black',
    borderRadius: 20,
    width: '80%',
    marginTop: 20,
  },
});

const img = StyleSheet.create({
  perfil:{
    borderRadius: 50,
    width: 110,
    height: 120,
  },
})