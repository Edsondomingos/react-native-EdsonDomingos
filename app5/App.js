import { View } from 'react-native';
//Pode ser qualquer nome inves de cabeçalho, pq a importação é default
//Mas se a função exportada não for default, precisa importar pelo nome correto.
import Cabe from './src/components/Cabecalho';
import Menu from './src/components/Menu'
import Anonima from './src/components/Anonima'
import ArrowFuntion from './src/components/ArrowFuntion'
import ArrowFuntionMenor from './src/components/ArrowFuntionMenor'
import Rodape from './src/components/Rodape'
import styles from './src/css/styles/style'
import styleCabecalho from './src/css/styles/styleCabecalho'
import Alerta from './src/components/Alerta'

export default function App() {
  return (
    <View style={styles.container}>
      <Cabe titulo="Cabeçalho" subtitulo="Subtitulo" style={styleCabecalho.cabecalho}/>
      <Menu titulo="Menu"/>
      <Alerta />
      <Anonima titulo="Anonima" />
      <ArrowFuntion titulo="ArrowFunction"/>
      <ArrowFuntionMenor titulo="ArrowFunctionMenor"/>
      <Rodape titulo="Rodape"/>
    </View>
  );
}