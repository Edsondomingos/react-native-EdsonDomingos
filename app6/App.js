import { View } from 'react-native';
import Stack from './src/navigation/Stack'
import { NavigationContainer } from '@react-navigation/native'

export default function App() {
  return (
    <View>
      <NavigationContainer>
        <Stack />
      </NavigationContainer>      
    </View>
  );
}
