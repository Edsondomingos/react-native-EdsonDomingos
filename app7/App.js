import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack'
import Index from './src/pages/Index'
import Contato from './src/pages/Contato'
import Sobre from './src/pages/Sobre'

//https://reactnavigation.org/docs/native-stack-navigator

const Stack = createNativeStackNavigator()


export default function App() {
  return (
    <NavigationContainer>
      <Stack.Navigator  initialRouteName='Index' screenOptions={{headerShown: false}}>
        {/* Quando definido com screen, automaticamente Index(screen) ja ganha propriedades(props)  */}
        <Stack.Screen name='Index' component={Index} options={{
            title: 'Home',
            headerStyle: {
              backgroundColor: 'purple'
            }
          }} />
        <Stack.Screen name='Contato' component={Contato} options={{title: 'Contato'}}/>
        <Stack.Screen name='Sobre' component={Sobre} />
      </Stack.Navigator>
    </NavigationContainer>
  )
}