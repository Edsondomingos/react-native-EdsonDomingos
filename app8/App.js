import { NavigationContainer } from "@react-navigation/native";
import { createNativeStackNavigator } from "@react-navigation/native-stack";
import Index from './src/pages/Index'
import Resultado from './src/pages/Resultado'

const Stack = createNativeStackNavigator()

export default () => {
  return (
    <NavigationContainer>
      <Stack.Navigator initialRouteName="Index">
        <Stack.Screen name="Index" component={Index}/>
        <Stack.Screen name="Resultado" component={Resultado}/>
      </Stack.Navigator>
    </NavigationContainer>
  )
}