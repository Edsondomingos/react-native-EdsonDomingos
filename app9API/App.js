import { NavigationContainer } from '@react-navigation/native';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import Index from './src/pages/Index';
import Cep from './src/pages/Cep';
import Temperatura from './src/pages/Temperatura';

const Tab = createBottomTabNavigator()

export default function App() {
  return (
    <NavigationContainer>
      <Tab.Navigator initialRouteName='Temperatura'>
        <Tab.Screen name='Index' component={Index} />
        <Tab.Screen name='Cep' component={Cep} />
        <Tab.Screen name='Temperatura' component={Temperatura} />
      </Tab.Navigator>
    </NavigationContainer>
  );
}
