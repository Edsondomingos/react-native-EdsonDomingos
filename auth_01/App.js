import { useState } from 'react';
import { Text, View, TextInput, Button, StyleSheet } from 'react-native';
import { app2 } from './src/config/firebaseconfig'
import { getAuth, createUserWithEmailAndPassword, signInWithEmailAndPassword, signOut   } from "firebase/auth";

export default () => {
  const [email, setEmail] = useState('')
  const [senha, setSenha] = useState('')

  function cadastrar(){
    const auth = getAuth(app2);
    createUserWithEmailAndPassword(auth, email, senha)
      .then((userCredential) => {
        // Signed in
        const user = userCredential.user;
        // ...
        setEmail('')
        setSenha('')
      })
      .catch((error) => {
        const errorCode = error.code;
        const errorMessage = error.message;
        // ..
        alert(errorMessage)
      });
  }

  function autenticar(){
    const auth = getAuth(app2);
    signInWithEmailAndPassword(auth, email, senha)
      .then((userCredential) => {
        // Signed in
        const user = userCredential.user;
        // ...
      })
      .catch((error) => {
        const errorCode = error.code;
        const errorMessage = error.message;
        alert(errorMessage)
      });
  }

  function desconectar(){
    const auth = getAuth(app2);
    signOut(auth).then(() => {
      // Sign-out successful.
      alert('Desconectado')
    }).catch((error) => {
      // An error happened.
      alert(errorMessage)
    });
  }

  function Verificar(){
    const auth = getAuth(app2);
    const user = auth.currentUser;
    if (user)
      alert(user.email)
    else
      alert('NÃ£o tem niguem autenticado')
  }

  return (
    <View style={styles.container}>
      <TextInput 
        style={styles.input}
        placeholder='E-mail' 
        onChange={setEmail}/>
      <TextInput 
        style={styles.input}
        placeholder='Senha' 
        secureTextEntry={true}
        onChange={setSenha}/>

      <Button title='Cadastrar' onPress={cadastrar}/>
      <Button title='Autenticar' onPress={autenticar} />
      <Button title='Verificar USuario Autenticado' onPress={Verificar}/>
      <Button title='Logout' onPress={desconectar}/>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    // width: 250,
  },
  input: {
    padding: 10,
    borderWidth: 1,
    margin: 5,
  },
})