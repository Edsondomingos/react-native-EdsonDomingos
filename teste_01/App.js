import { Button, StyleSheet, Text, TextInput, View } from 'react-native';

export default function App() {
  return (
    <View>
      
      <Text testID='TextTitulo' style={styles.titulo}>Digite um numero</Text>

      <TextInput 
        style={styles.entrada}
        placeholder='Digite aqui'
        testID='TextInputEntrada'
      />

      <Button testID='Button' title="Calcular"/>

    </View>
  );
}

const styles = StyleSheet.create({
  titulo: {
    textAlign: 'center',
    fontSize: 30,
  },
  entrada: {
    borderBottomWidth: 1,
    margin: 10,
    padding: 10,
    outline: 0,
  },
})