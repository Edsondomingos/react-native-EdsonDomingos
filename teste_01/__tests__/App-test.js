import renderer from "react-test-renderer" //Biblioteca que o jest usa
import App from '../App'
import {render}  from "@testing-library/react-native"

describe('Teste de Snapshot', () => {
    it('Criação de snapshot', () => {
        const inicial = renderer.create(<App />).toJSON()
        expect(inicial).toMatchSnapshot();
    })
})

describe('Teste de inicialização', () => {
    it('Verificando se componentes foram iniciados', () => {
        // const [getByText, getByTextId, getAllByText] = render(<App />)
        const {getByText, getByTestId, getAllByText} = render(<App />)

        const tituloApp = getByTestId('TextTitulo')
        expect(tituloApp).toBeDefined() // Verifica se foi inicializado e se esta la na tela

        expect(getByTestId('TextInputEntrada')).toBeDefined()

        const botao = getByText('Calcular')
        expect(botao).toBeDefined()
    })
})