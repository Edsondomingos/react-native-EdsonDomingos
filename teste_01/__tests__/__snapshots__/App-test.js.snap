// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teste de Snapshot Criação de snapshot 1`] = `
<View>
  <Text
    style={
      Object {
        "fontSize": 30,
        "textAlign": "center",
      }
    }
    testID="TextTitulo"
  >
    Digite um numero
  </Text>
  <TextInput
    placeholder="Digite aqui"
    style={
      Object {
        "borderBottomWidth": 1,
        "margin": 10,
        "outline": 0,
        "padding": 10,
      }
    }
    testID="TextInputEntrada"
  />
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="Button"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Calcular
      </Text>
    </View>
  </View>
</View>
`;
